ibis/common/selectors.py

```diff
@@ -29,3 +28,79 @@ def expand(self, table: ir.Table) -> Sequence[ir.Value]:
             A sequence of value expressions that match the selector
 
         """
+
+    @abc.abstractmethod
+    def expand_names(self, table: ir.Table) -> frozenset[str]:
+        """Compute the set of column names that match the selector."""
+
+
+class Selector(Expandable):
```

Comment:
This now looks a lot more like the selectors implementation I wrote for ibis-ml (https://github.com/ibis-project/ibis-ml/blob/main/ibis_ml/select.py) (there's _mild_ reasons why that project needed it's own implementation). Not sure if there's anything worth cribbing from there, but figured I'd raise it.

Reply:
`__sub__` looks interesting, I'm guessing that's supported in tidyr/dplyr?

Reply:
Maybe (can't remember)? Selecting "things that match one thing, excluding things that match another thing" was something I was doing a lot of, so the short-hand was also nice.

Reply:
Will do in a follow up. I also want to move everything into a single file, it's really annoying having two files for the implementation here.